_drafts/installing-gdb.md

```diff
@@ -0,0 +1,514 @@
+---
+title: "Tools we use: installing `gdb` for ARM"
+description: Demonstrate a few methods of installing gdb for ARM
+author: noah
+image: img/<post-slug>/cover.png # 1200x630
+---
+
+<!-- excerpt start -->
+
+In this mini article, I'll be going on a few strategies and nuances around
+getting a copy of GDB installed for debugging ARM chips.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## GDB for ARM
+
+GDB, the GNU Project Debugger, is by default typically compiled to target the
+same architecture as the host system. When we talk about "host" and "target
+architectures, what we mean is:
+
+1. the **host** architecture: where the GDB program itself is run
+2. the **target** architecture: where the program being debugged is run
+
+Debugging a program that's running on the same machine as the host might look
+like this:
+
+[![img/installing-gdb/gdb_host_target.drawio.svg](/img/installing-gdb/gdb_host_target.drawio.svg)](/img/installing-gdb/gdb_host_target.drawio.svg)
+
+In embedded engineering, we often want to target a foreign architecture (eg, the
+embedded device, connected to some debug probe). That setup might look like
+this:
+
+[![img/installing-gdb/gdb_cross_target.drawio.svg](/img/installing-gdb/gdb_cross_target.drawio.svg)](/img/installing-gdb/gdb_cross_target.drawio.svg)
+
+We need a version of GDB that supports the target architecture of the program
+being debugged.
+
+## Summary of strategies
+
+I'm going to go through the various ways to install GDB with ARM support, but
+first here's a table summarizing the approaches:
+
+| Strategy                   | Binary or source install? | Pinned to version | Python support        |
```

Comment:
An additional column might be "requires a 3rd party package manager?" where Conda, Docker, ... are yes